master/internal/stream/system.go

```diff
@@ -158,13 +237,21 @@ func (ps PublisherSet) Websocket(socket *websocket.Conn, c echo.Context) error {
 	//   - fallin
 	//   - fallout
 	//
-	// (note that online appearences and disappearances are not supported; we'll detect those
-	// situations and just break the connection to the relevant streaming clients).
+	// (note that online appearances and disappearances are not supported; we'll detect those
+	// situations and just break the connection to streaming clients).
 
 	// detect context cancelation, and bring it into the websocket thread
 	go func() {
-		<-ctx.Done()
-		streamer.Close()
+		select {
+		case <-ctx.Done():
+			streamer.Close()
+		// XXX: is this redundant?
+		case <-ssupCtx.Done():
+			// XXX: if this isn't redundant, is this the correct response?
```

Comment:
Yes, regardless of what scope the cancelation occurs at, the right response for this particular streamer is to shut down.